tests/MongoDB.Driver.Tests/AsyncCursorTests.cs

```diff
@@ -53,6 +56,54 @@ public void Cursor_should_not_throw_exception_after_double_close([Values(false,
             }
         }
 
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void KillCursor_should_actually_work(
+            [Values(false, true)] bool async)
+        {
+            RequireServer.Check().Supports(Feature.KillCursorsCommand);
+            var eventCapturer = new EventCapturer().Capture<CommandSucceededEvent>(x => x.CommandName.Equals("killCursors"));
+            using (var client = DriverTestConfiguration.CreateDisposableClient(eventCapturer))
+            {
+                IAsyncCursor<BsonDocument> cursor;
+                var database = client.GetDatabase("admin");
+                var collection = database.GetCollection<BsonDocument>(GetType().Name);
+                if (async)
+                {
+                    for (int i = 0; i < 1000; i++)
+                    {
+                        collection.InsertOneAsync(new BsonDocument("x", i)).GetAwaiter().GetResult();
+                    }
+
+                    cursor = collection.FindAsync("{}").GetAwaiter().GetResult();
+                    cursor.MoveNextAsync().GetAwaiter().GetResult();
+                }
+                else
+                {
+                    for (int i = 0; i < 1000; i++)
+                    {
+                        collection.InsertOne(new BsonDocument("x", i));
+                    }
+
+                    cursor = collection.FindSync("{}");
+                    cursor.MoveNext();
+
+                }
+
+                long cursorId;
+                using (cursor)
```

Comment:
Anyone else find this a confusing use of the `using` statement?  I usually see it used (in the Java equivalent) where the used object is also created in the using statement itself, not up above.  Maybe better to just call Dispose explicitly here, since that's the method that is under test.

CC @vincentkam @DmitryLukyanov

Reply:
Discussed offline with Katie!